lib/ansible/cli/galaxy.py

```diff
@@ -911,68 +918,95 @@ def execute_install(self):
-            collections_path = C.COLLECTIONS_PATHS
+        two_type_warning = "The requirements file '%s' contains {0}s which will be ignored. To install these {0}s " \
+                           "run 'ansible-galaxy {0} install -r' or to install both at the same time run " \
+                           "'ansible-galaxy install -r' without a custom install path." % to_text(requirements_file)
 
-            if len([p for p in collections_path if p.startswith(output_path)]) == 0:
-                display.warning("The specified collections path '%s' is not part of the configured Ansible "
-                                "collections paths '%s'. The installed collection won't be picked up in an Ansible "
-                                "run." % (to_text(output_path), to_text(":".join(collections_path))))
+        # TODO: Would be nice to share the same behaviour with args and -r in collections and roles.
+        collection_requirements = []
+        role_requirements = []
+        if context.CLIARGS['type'] == 'collection':
+            collection_path = GalaxyCLI._resolve_path(context.CLIARGS['collections_path'])
+            requirements = self._require_one_of_collections_requirements(install_items, requirements_file)
 
-            output_path = validate_collection_path(output_path)
-            b_output_path = to_bytes(output_path, errors='surrogate_or_strict')
-            if not os.path.exists(b_output_path):
-                os.makedirs(b_output_path)
+            collection_requirements = requirements['collections']
+            if requirements['roles']:
+                display.display(two_type_warning.format('role'))
+        else:
+            if not install_items and requirements_file is None:
+                raise AnsibleOptionsError("- you must specify a user/role name or a roles file")
 
-            install_collections(requirements, output_path, self.api_servers, (not ignore_certs), ignore_errors,
-                                no_deps, force, force_deps, context.CLIARGS['allow_pre_release'])
+            if requirements_file:
+                if not (requirements_file.endswith('.yaml') or requirements_file.endswith('.yml')):
+                    raise AnsibleError("Invalid role requirements file, it must end with a .yml or .yaml extension")
+
+                requirements = self._parse_requirements_file(requirements_file)
+                role_requirements = requirements['roles']
+
+                # We can only install collections and roles at the same time if the type wasn't specified and the -p
+                # argument was not used. If collections are present in the requirements then at least display a msg.
+                galaxy_args = self._raw_args
+                if requirements['collections'] and (not self._implicit_role or '-p' in galaxy_args or
+                                                    '--roles-path' in galaxy_args):
+                    display.display(two_type_warning.format('collection'))
```

Comment:
I wonder if this will end up feeling too noisy since there's no way to suppress it for people intentionally using a combined requirements.yml with the explicit `role`/`collection` subcommand + custom path. In those two cases, maybe this could be display.vvv output?

Reply:
That's a good point, maybe we should have this output to `vvv` when using the explicit role/collection subcommand but add it to the warning when no explicit role|collection was specified. That way people who do `ansible-galaxy install -r requirements.yml -p ./` will still be informed that the collections were skipped. People who specify `ansible-galaxy collection install` or `ansible-galaxy role install` should already expect it will only install those types of resources.

@geerlingguy what do you think?

Reply:
I've ended up taking your suggestion and hiding these messages behind `-vvv`. The `ansible-galaxy` cli will still warn when `ansible-galaxy install -r ... -p ...` is used though as I see people definitely getting confused there if we were to just skip collections.